packages/waku/src/router/client.ts

```diff
@@ -166,6 +168,32 @@ export function Link({
         throw new Error('Missing Router');
       };
   const [isPending, startTransition] = useTransition();
+  const ref = useRef<HTMLAnchorElement>(null);
+
+  useEffect(() => {
+    if (unstable_prefetchOnView && ref.current) {
+      const observer = new IntersectionObserver(
+        (entries) => {
+          entries.forEach((entry) => {
+            if (entry.isIntersecting) {
+              const url = new URL(to, window.location.href);
+              if (url.href !== window.location.href) {
+                const route = parseRoute(url);
+                prefetchRoute(route);
+              }
+            }
+          });
+        },
+        { threshold: 0.1 },
+      );
+
+      observer.observe(ref.current);
+
+      return () => {
+        observer.disconnect();
+      };
+    }
+  }, [unstable_prefetchOnView, prefetchRoute, to, ref]);
```

Comment:
`ref` isn't necessary.
`prefetchRoute` is not stable, you want just `router` for deps.